glide-core/src/client/value_conversion.rs

```diff
@@ -379,9 +369,99 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // `FUNCTION LIST` returns an array of maps with nested list of maps.
+        // In RESP2 these maps are represented by arrays - we're going to convert them.
+        /* RESP2 response
+        1) 1) "library_name"
+           2) "mylib1"
+           3) "engine"
+           4) "LUA"
+           5) "functions"
+           6) 1) 1) "name"
+                 2) "myfunc1"
+                 3) "description"
+                 4) (nil)
+                 5) "flags"
+                 6) (empty array)
+              2) 1) "name"
+                 ...
+        2) 1) "library_name"
+           ...
+
+        RESP3 response
+        1) 1# "library_name" => "mylib1"
+           2# "engine" => "LUA"
+           3# "functions" =>
+              1) 1# "name" => "myfunc1"
+                 2# "description" => (nil)
+                 3# "flags" => (empty set)
+              2) 1# "name" => "myfunc2"
+                 ...
+        2) 1# "library_name" => "mylib2"
+           ...
+        */
+        ExpectedReturnType::ArrayOfMapsRecursive => match value {
+            // empty array, or it is already contains a map (RESP3 response) - no conversion needed
+            Value::Array(ref array) if array.is_empty() || matches!(array[0], Value::Map(_)) => {
+                Ok(value)
+            }
+            Value::Array(array) => {
+                convert_array_of_flat_maps(array, Some(ExpectedReturnType::ArrayOfMaps))
+            }
+            // cluster (multi-node) response - go recursive
+            Value::Map(map) => convert_map_entries(
+                map,
+                Some(ExpectedReturnType::BulkString),
+                Some(ExpectedReturnType::ArrayOfMapsRecursive),
+            ),
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted into a recursive array of maps",
+            )
+                .into()),
+        },
+        // Not used for a command, but used as a helper for `ArrayOfMapsRecursive` to process the inner map.
+        ExpectedReturnType::ArrayOfMaps => match value {
+            Value::Array(array) => {
+                convert_array_of_flat_maps(array, Some(ExpectedReturnType::StringOrSet))
+            }
+            // Due to recursion, it would be called to convert every map value, including simple strings, do nothing with them
```

Comment:
I would reword this comment to make it easier to read. I think this would be fine:
```suggestion
            // Due to recursion, this will convert every map value, including simple strings, which we do nothing with
```